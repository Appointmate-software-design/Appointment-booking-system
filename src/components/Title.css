.header{
    position: relative;
    text-align: center;
    background-color:rgba(82, 82, 213, 1);
    padding: 2rem;
    box-shadow:1px 2px 3px 0px rgba(0, 0, 0, 0.5) ;
}

.title{
    font-size: 40px;
    color: rgba(255, 255, 255, 1);
    font-weight: bolder;
    text-align: center;
}

.logo{
    width: 110px;
    height: auto;
    position: absolute;
    margin-left: 3px;
    top: 0.5rem;
    left: 1rem;
}