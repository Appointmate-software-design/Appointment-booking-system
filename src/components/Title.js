import React from 'react'

// This component is the Title component and appears on every page.
// It contains the logo and the word "appointmate", the name of our app.
export default function Title() {
  return (
    <div
      style={{
        padding: '2rem',
        position: 'relative',
        textAlign: 'center'
      }}
    >
      <h1
        style={{
          // Set the background color gradient to the text
          backgroundImage: 'linear-gradient(to bottom, #0077B5, #FFFFFF)',
          // Set the Webkit background clip to text to show the gradient
          WebkitBackgroundClip: 'text',
          // Set the text color to transparent to show the gradient
          WebkitTextFillColor: 'transparent',
          fontSize: '70px'
        }}
      >
        appointmate
      </h1>
      <img
        src={process.env.PUBLIC_URL + '/logo-appointmate.png'}
        alt="Appointmate Logo"
        style={{
          width: '100px',
          height: 'auto',
          position: 'absolute',
          top: '1rem',
          left: '1rem'
        }}
      />
    </div>
  );
}
