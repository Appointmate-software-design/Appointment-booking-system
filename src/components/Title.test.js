import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Title from './Title';
import renderer from 'react-test-renderer';

describe('Title component', () => {
  beforeEach(() => {
    render(<Title />);
  });

  test('renders the Appointmate text', () => {
    const titleElement = screen.getByText(/appointmate/i);
    expect(titleElement).toBeInTheDocument();
  });

  test('renders the Appointmate logo', () => {
    const logoElement = screen.getByAltText(/Appointmate Logo/i);
    expect(logoElement).toBeInTheDocument();
    expect(logoElement).toHaveAttribute('src', expect.stringContaining('logo-appointmate.png'));
  });

  test('Appointmate text has gradient style', () => {
    const titleComponent = renderer.create(<Title />).toJSON();
    expect(titleComponent).toMatchSnapshot();
  });
});
