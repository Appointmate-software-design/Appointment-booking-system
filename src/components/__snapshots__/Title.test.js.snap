// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component Appointmate text has gradient style 1`] = `
<div
  style={
    Object {
      "padding": "2rem",
      "position": "relative",
      "textAlign": "center",
    }
  }
>
  <h1
    style={
      Object {
        "WebkitBackgroundClip": "text",
        "WebkitTextFillColor": "transparent",
        "backgroundImage": "linear-gradient(to bottom, #0077B5, #FFFFFF)",
        "fontSize": "70px",
      }
    }
  >
    appointmate
  </h1>
  <img
    alt="Appointmate Logo"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    src="/logo-appointmate.png"
    style={
      Object {
        "height": "auto",
        "left": "1rem",
        "position": "absolute",
        "top": "1rem",
        "transform": "none",
        "transition": "transform 0.3s",
        "width": "100px",
      }
    }
  />
</div>
`;
